cmd/thanos/store.go

```diff
@@ -55,6 +55,10 @@ func registerStore(m map[string]setupFunc, app *kingpin.Application) {
 		"YAML file that contains index cache configuration. See format details: https://thanos.io/components/store.md/#index-cache",
 		false)
 
+	cachingBucketConfig := extflag.RegisterPathOrContent(cmd, "experimental.caching-bucket.config",
```

Comment:
I know we were doing opposite but now when I think of it I think it would be nice to have this is hidden but normal flag name. 

It's painful to change flag names for users ... (: 

so maybe just 
```suggestion
	cachingBucketConfig := extflag.RegisterPathOrContent(cmd, "store.caching-bucket.config",
```

Reply:
How can we hide it?

Reply:
I guess by adding `RegisterHiddenPathOrContent` or something (:

Reply:
Oh, are you suggesting to add `RegisterHiddenPathOrContent` that will "hide" the command line option – how? By giving it "hidden" prefix? I'm little confused about what you're suggesting here.

Reply:
Hidden by using new `extflag.HiddenCmdClause(cmd)` like this:

```
	cachingBucketConfig := extflag.RegisterPathOrContent(extflag.HiddenCmdClause(cmd), "store.caching-bucket.config", ...)
```

WDYT?